src/server/json_family.cc

```diff
@@ -490,8 +491,66 @@ OpResult<long> OpDel(const OpArgs& op_args, string_view key, string_view path) {
   return total_deletions;
 }
 
+OpResult<vector<StringVector>> OpObjKeys(const OpArgs& op_args, string_view key,
+                                         JsonExpression expression) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<StringVector> vec;
+  auto cb = [&vec](const string_view& path, const json& val) {
+    if (!val.is_object()) {
+      vec.emplace_back();
+      return;
+    }
+
+    auto& current_object = vec.emplace_back();
```

Comment:
what happens if you have a hierarchy of more than 2 levels?

Reply:
The parent is always stored before their descendants in the Vector. A parent can have multiple descendants if the JsonPath is recursive and the parent index within the vector is smaller than the descendant index.